### PR TITLE
Ensure the disk is free to use after seed.apply

### PR DESCRIPTION
**Pull request by cbosdo**
# What does this PR do?

`seed.apply` triggers a `guestmount` of the disk file image. To properly
unmount the image we need to call `guestunmount` and then wait for the
disk to be free.

### What issues does this PR fix or reference?

issue #56002

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by cbosdo** [comment]
just rebased on top of master

**Comment by cbosdo** [comment]
Rebased on master.

**Review by DmitryKuzmenko — DISMISSED** [review]
Approving in assumption the typo will be fixed. See inline comment.

File: salt/modules/guestfs.py [hunk 1]
```diff
@@ -71,3 +72,23 @@ def mount(location, access='rw', root=None):
...
+    '''
+    Unmount an image
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' guest.unmount /mountpoint disk=/srv/images/fedora.qcow
+    '''
+    cmd = 'guestunmount {0}'.format(name)
```
Comment (DmitryKuzmenko) [inline salt/modules/guestfs.py:87]: Maybe it's also a good idea to put `-q` argument here since we're not handling the output.

**Comment by DmitryKuzmenko** [comment]
Probably it's also a good idea to add a way to provide the `guestunmount --retry` count and the `lsof` waiting timeout from outside but since it's easily TBD and not used in the code I'm okay with the current code.

File: salt/modules/guestfs.py (hunk 1 above)
Comment (cbosdo) [reply salt/modules/guestfs.py:87]: right, tough `guestunmount` is not the most verbose tool :)

File: salt/modules/guestfs.py [hunk 2]
```diff
@@ -71,3 +72,23 @@ def mount(location, access='rw', root=None):
...
+
+def umount(name, disk=None):
+    '''
+    Unmount an image
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' guest.unmount /mountpoint disk=/srv/images/fedora.qcow
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply salt/modules/guestfs.py:85]: nice catch

[the remark this answers is not in the document]
Comment (cbosdo) [reply salt/modules/guestfs.py:85]: even `guestfs.umount`

**Comment by cbosdo** [comment]
> Probably it's also a good idea to add a way to provide the `guestunmount --retry` count and the `lsof` waiting timeout from outside but since it's easily TBD and not used in the code I'm okay with the current code.

`--retry` could be of some help but wouldn't solve the original issue.

File: salt/modules/guestfs.py (hunk 1 above)
Comment (cbosdo) [reply salt/modules/guestfs.py]: fixed

File: salt/modules/guestfs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbosdo) [reply salt/modules/guestfs.py]: fixed

**Comment by cbosdo** [comment]
Added `-q` parameter to `guestunmount` call and fixed typo

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by DmitryKuzmenko** [comment]
Thank you! Let's wait for tests.

**Comment by cbosdo** [comment]
rebased on top of master (post Blackening commit)

**Comment by DmitryKuzmenko** [comment]
Now it should be good. Thank you @cbosdo

**Pull request merged**